seqr/views/apis/summary_data_api.py

```diff
@@ -172,6 +183,140 @@ def bulk_update_family_analysed_by(request):
     })
 
 
+def _load_aip_data(data, user):
+    category_map = data['metadata']['categories']
+    results = data['results']
+
+    family_id_map = dict(Individual.objects.filter(
+        family__project__in=get_internal_projects(), individual_id__in=results.keys(),
+    ).values_list('individual_id', 'family_id'))
+    missing_individuals = set(results.keys()) - set(family_id_map.keys())
+    if missing_individuals:
+        raise ErrorsWarningsException([f'Unable to find the following individuals: {", ".join(sorted(missing_individuals))}'])
+
+    all_variant_ids = set()
+    family_variant_data = {}
+    for family_id, variant_pred in results.items():
+        family_variant_data.update({
+            (family_id_map[family_id], variant_id): pred for variant_id, pred in variant_pred.items()
+        })
+        all_variant_ids.update(variant_pred.keys())
+
+    saved_variant_map = {
+        (v.family_id, v.variant_id): v
+        for v in SavedVariant.objects.filter(family_id__in=family_id_map.values(), variant_id__in=all_variant_ids)
+    }
+
+    new_variants = set(family_variant_data.keys()) - set(saved_variant_map.keys())
+    if new_variants:
+        saved_variant_map.update(_search_new_saved_variants(new_variants, user))
+
+    aip_tag_type = VariantTagType.objects.get(name=AIP_TAG_TYPE, project=None)
+    existing_tags = {
+        tuple(t.saved_variant_ids): t for t in VariantTag.objects.filter(
+            variant_tag_type=aip_tag_type, saved_variants__in=saved_variant_map.values(),
+        ).annotate(saved_variant_ids=ArrayAgg('saved_variants__id', ordering='id'))
+    }
+
+    today = datetime.now().strftime('%Y-%m-%d')
```

Comment:
Do we typically do dates in ET or UTC?

Reply:
Nope! We probably should, but that would be an out of scope change all over the repo